proto/commons/massa/model/v1/commons.proto

```diff
@@ -34,3 +34,19 @@ message SecureShare {
   // A secure hash of the data. See also [massa_hash::Hash]
   string id = 5;
 }
+
+// Comparison result
+message ComparisonResult {
+  // Left is lower
+  message LeftLower {}
+  // Left is equal to right
+  message Equal {}
+  // Left is greater
+  message LeftGreater {}
+  // Status
+  oneof restult {
```

Comment:
out of curiosity why not an Enum to avoid having to declare 3 types ?

Reply:
General rule: when there are multiple options, enum unless some of those options need to be complex types, then use OneOf

Reply:
typo: restult

Reply:
done :)

Reply:
cool ! can you check if it was also done elsewhere ?

Reply:
I'm currently applying modifications to other repositories.
I'll also check scheme previously used for ComparisonResult does not appear elsewhere

Reply:
Haven't found any other "creative" use of OneOf in the .protos present in this branch ^^